### PR TITLE
chore: Perform feature/config condition checks at request time

### PR DESCRIPTION
**Pull request by benjreinhart**
### SUMMARY

This builds on https://github.com/apache/superset/pull/14568 and https://github.com/apache/superset/pull/14650 by swapping out the remaining occurrences of code that conditionally registers resources during startup. It should now be possible to flip config/feature values on/off without restarting the server.

### TEST PLAN

I enabled some disabled tests and added many more. I also tested many of these endpoints manually.

cc @robdiciuccio @craig-rueda @dpgaspar

### CODE REVIEW

File: superset/connectors/druid/views.py
Comment (villebro) [inline superset/connectors/druid/views.py:54]: nit: could this be called `EnsureEnabledMixin`?

File: superset/connectors/druid/views.py
Comment (villebro) [inline superset/connectors/druid/views.py:65]: nit: while it really doesn't matter here, `EnsureEnabled` should probably come to the left of `SupersetModelView` as per convention (same for the others).

File: superset/connectors/sqla/views.py
Comment (villebro) [inline superset/connectors/sqla/views.py:381]: We could potentially generalize this by breaking this out into a Mixin that checks `is_feature_enabled` on a reserved property. Something like this:

```python
class FeatureFlagMixin:
    feature_flag: str

    def is_enabled(self) -> bool:
        return is_feature_enabled(self.feature_flag)

    @before_request
    def ensure_enabled(self) -> None:
        if not self.is_enabled():
            raise NotFound()
```

Then later this could be used as follows:

```python
class RowLevelSecurityFiltersModelView(  # pylint: disable=too-many-ancestors
    FeatureFlagMixin, SupersetModelView, DeleteMixin
):
    feature_flag = "ROW_LEVEL_SECURITY"
    ....
```
we could potentially also add support for callbacks that make it possible to add custom validators that don't rely on `is_feature_enabled`:

```python
class FeatureFlagMixin:
    feature_flag: Union[str, Callable[[], str]]

    def is_enabled(self) -> bool:
        if callable(self.feature_flag):
            return self.feature_flag()
        return is_feature_enabled(self.feature_flag)

    @before_request
    def ensure_enabled(self) -> None:
        if not self.is_enabled():
            raise NotFound()
```

and then use it like this:

```python
    feature_flag = lambda _: app.config["FAB_ADD_SECURITY_VIEWS"] and app.config["SUPERSET_LOG_VIEW"]
```

**Review by villebro — COMMENTED** [review]
Exciting stuff! Left a few comments/ideas

File: superset/connectors/sqla/views.py
Comment (benjreinhart) [reply superset/connectors/sqla/views.py:381]: I think we could, but it's a little bit more complicated then that. `@before_request` needs to be able to be scoped to a subset of route handlers in some cases. We can achieve that, but for now I think we should ship the more naive version. If this pattern continues to be used moving forward, the opportunity for a refactor should present itself.

**Comment by benjreinhart** [comment]
@villebro Thank you for the solid feedback as always! I updated the naming and moved the order of inheritance around per the convention. If you're ok with it, I think my preference would be to hold off on a refactor around the mixin for now. Let me know what you think.

File: tests/alerts_tests.py
Comment (robdiciuccio) [inline tests/alerts_tests.py:382]: Nit: might be better to do a positive assertion (200) on these.

**Review by robdiciuccio — APPROVED** [review]

File: tests/alerts_tests.py
Comment (benjreinhart) [reply tests/alerts_tests.py:382]: Yes I agree, unfortunately many of these return 302s since I believe more setup is required to get the right users access to the resources. Ideally, these tests would do that, but seeing as I wasn't able to resolve it right away, I wanted to get this PR out and also keep the changes to a minimum.

Comment (robdiciuccio) [reply tests/alerts_tests.py:382]: How about `self.assertLess(rv.status_code, 400)` in that case?

Comment (benjreinhart) [reply tests/alerts_tests.py:382]: That works for me 👍🏻

**Review by robdiciuccio — APPROVED** [review]

File: superset/connectors/sqla/views.py
Comment (villebro) [reply superset/connectors/sqla/views.py:381]: Gotcha - I don't have the full picture of the context in which this change is used, so some of my comments will probably be irrelevant 👍

**Review by villebro — APPROVED** [review]
LGTM 👍

**Review by dpgaspar — COMMENTED** [review]
LGTM, after tests pass

File: superset/app.py
Comment (benjreinhart) [reply superset/app.py:233]: I'm not sure, but this should maintain parity with the previous code.

**Comment by vedangparasnis** [comment]
What will be the json payload format required for rowlevelsecurityfiltersmodelview/api/create . I can see there is an api for rowlevelsecurityfiltersmodelview/api/, however, can this api be used to create a new row level security using api. 

Thanks in advance

**Pull request merged**
